code/controllers/subsystem/speech_controller.dm

```diff
@@ -0,0 +1,64 @@
+SUBSYSTEM_DEF(speech_controller)
+	name = "Speech Controller"
+	wait = 1
+	flags = SS_TICKER
+	priority = FIRE_PRIORITY_SPEECH_CONTROLLER//has to be high priority, second in priority ONLY to SSinput
+	init_order = INIT_ORDER_SPEECH_CONTROLLER
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
+
+	///associative list of the form: list(client mob = message or messages that mob is queued to say).
+	///this is our process queue, processed every tick.
+	var/list/queued_says_to_execute = list()
+	///how many says we've processed since the last stat_entry()
+	var/say_tracker = 0
+
+///queues mob_to_queue into our process list so they say(message) near the start of the next tick
+/datum/controller/subsystem/speech_controller/proc/queue_say_for_mob(mob/mob_to_queue, message)
+	if(QDELETED(mob_to_queue) || !message)
+		return FALSE
```

Comment:
Why qdel/!message check? isn't that already done by users?

Reply:
they could be deleted inbetween the say_verb() and start of the subsystems fire

Reply:
why is that a problem? Was the case with the old system too

Reply:
oh i misread what part the review was in. its for safety even though i doubt someone is going to put that into a situation where its necessary :clueless: